apps/content-gateway-api/src/app/app.ts

```diff
@@ -1,98 +1,98 @@
-} from "@banklessdao/content-gateway-sdk";
+import { createLogger } from "@banklessdao/util-misc";
 import {
     ContentGateway,
+    ContentGatewayUser,
     createContentGateway,
     DataRepository,
+    UserRepository
 } from "@domain/feature-gateway";
-import { createLogger, programError } from "@banklessdao/util-misc";
 import * as express from "express";
 import { graphqlHTTP } from "express-graphql";
 import * as g from "graphql";
-import { MongoClient } from "mongodb";
+import { Collection, MongoClient, ObjectId } from "mongodb";
 import { join } from "path";
-import { Logger } from "tslog";
 import {
-    createGraphQLAPIV1 as createGraphQLAPIV1,
-    createInMemoryOutboundDataAdapter,
+    authorization,
+    createGraphQLAPIV1,
+    createMongoUserRepository,
     ObservableSchemaRepository,
-    toObservableSchemaRepository,
+    toObservableSchemaRepository
 } from ".";
 import { createMongoDataRepository, createMongoSchemaRepository } from "./";
 import { liveLoaders } from "./live-loaders";
 import { LiveLoader } from "./live-loaders/LiveLoader";
+import { MongoUser } from "./repository/mongo/MongoUser";
 import { generateContentGatewayAPIV1 } from "./service";
 
 export type ApplicationContext = {
-    logger: Logger;
-    env: string;
-    isDev: boolean;
-    isProd: boolean;
     app: express.Application;
-    mongoClient: MongoClient;
     schemaRepository: ObservableSchemaRepository;
+    userRepository: UserRepository;
     dataRepository: DataRepository;
     contentGateway: ContentGateway;
-    client: ContentGatewayClientV1;
 };
 
-export const createApp = async ({
-    dbName,
-    mongoClient,
-}: {
+export type AppParams = {
+    nodeEnv: string;
+    resetDb: boolean;
+    addFrontend: boolean;
     dbName: string;
     mongoClient: MongoClient;
-}) => {
-    const env = process.env.NODE_ENV ?? programError("NODE_ENV not set");
-    const isDev = env === "development";
-    const isProd = env === "production";
-    const resetDb = process.env.RESET_DB === "true";
-    const addFrontend = process.env.ADD_FRONTEND === "true";
+    schemasCollectionName: string;
+    usersCollectionName: string;
+    rootUser: ContentGatewayUser;
+    rootApiKey: string;
+};
 
+export const createApp = async (params: AppParams) => {
     const logger = createLogger("ContentGatewayAPIApp");
+    const { dbName, mongoClient } = params;
+    const db = mongoClient.db(dbName);
+    const users = db.collection<MongoUser>(params.usersCollectionName);
 
-    if (resetDb) {
+    if (params.resetDb) {
         await mongoClient.db(dbName).dropDatabase();
     }
-    logger.info(`Running in ${env} mode`);
+    logger.info(`Running in ${params.nodeEnv} mode`);
 
     const app = express();
+
     const schemaRepository = toObservableSchemaRepository(
-        await createMongoSchemaRepository({ dbName, mongoClient })
+        await createMongoSchemaRepository({
+            db,
+            collName: params.schemasCollectionName,
```

Comment:
Now we can supply a name for both the users and the schemas collection. This was necessary because the integration tests were flapping (and might be significant later because of multi-tenancy).